packages/core/src/actors/callback.ts

```diff
@@ -134,36 +134,25 @@ export function fromCallback<TEvent extends EventObject, TInput = unknown>(
 ): CallbackActorLogic<TEvent, TInput> {
   const logic: CallbackActorLogic<TEvent, TInput> = {
     config: invokeCallback,
-    start: (_state, { self, system }) => {
-      system._relay(self, self, { type: 'xstate.create' });
```

Comment:
The previous version was somewhat more correct. At the moment, `start` should not be able to modify the snapshot. `transition` should also do it in an immutable way and even the previous version didn't do it "correctly" - but at least the mutation was located in a more appropriate place (it was easier to refactor it towards the immutable version than it is with the proposed change)